src/catalog/src/durable/debug.rs

```diff
@@ -43,8 +43,9 @@ pub trait Collection: Debug {
     /// Extract the [`CollectionTrace`] from a [`Trace`] that corresponds to [`Collection`].
     fn collection_trace(trace: Trace) -> CollectionTrace<Self>;
 
-    /// Return [`TypedCollection`] that corresponds to [`Collection`].
-    fn stash_collection() -> TypedCollection<Self::Key, Self::Value>;
+    /// If the collection has a corresponding stash implementation, return
+    /// [`TypedCollection`] that corresponds to [`Collection`].
+    fn stash_collection() -> Option<TypedCollection<Self::Key, Self::Value>>;
```

Comment:
If you're planning on waiting until the stash implementation is removed, then you'll probably have to revert this back to not being an `Option`.

Reply:
That's fine--will need to rebase on top of whatever changes exist.